docs: Update docs for better installation and provider setups (#408)
block/goose: docs/installation.md

# Installation

> To install Goose, use `pipx` - works on macOS, linux, and Windows. 

Reviewer: ```suggestion
To install Goose, use `pipx` on macOS, Linux, or Windows. 
```

[pipx]: https://github.com/pypa/pipx?tab=readme-ov-file#install-pipx

### Configuration

#### Set up a provider
Goose works with a set of [supported LLM providers][providers] that you can obtain an API key from if you don't already have one. You'll be prompted to set an API key if you haven't set one previously when you run Goose.

>[!TIP]
> **Billing:**
>
> You will need to have credits in your LLM Provider account (when necessary) to be able to successfully make requests.
>

#### Profiles

> After installation, you can configure Goose anytime by editing your profile file located at `~/.config/goose/profiles.yaml`. You can set multiple profile configurations, use different LLM providers, or enable toolkits that customize Goose's functionality as well:

Reviewer: ```suggestion
After installation, you can configure Goose anytime by editing your profile file located at `~/.config/goose/profiles.yaml`. You can set multiple profile configurations, use different LLM providers, and enable toolkits that customize Goose's functionality as well:
```
